### PR TITLE
Add test and fix for report which uses multiple files from same dependency

### PR DESCRIPTION
**Pull request by r-ash**
Spotted a new issue when migrating Debebe's orderly. If someone is pulling in 2 files from the same dependency, we are not handling this case in the migration.

I've added a test which recreates the problem and have fixed the impl

### CODE REVIEW

File: R/migrate.R [hunk 1]
```diff
@@ -249,14 +249,14 @@ archive_migrate_depends <- function(depends, parameters) {
   }
   ## Seen in rtm_incoming_serology/20200603-204022-70b8bfa5
   if (is.null(depends$index)) {
-    depends$index <- as.integer(factor(depends$id))
+    depends$index <- as.integer(factor(paste0(depends$id, depends$id_requested)))
```
Comment (r-ash) [inline R/migrate.R:252]: I don't *think* we should ever get this where they have the same ID, but have requested it in slightly different ways. But have put this in to be extra careful.

File: R/migrate.R [hunk 2]
```diff
@@ -249,14 +249,14 @@ archive_migrate_depends <- function(depends, parameters) {
...
   }
   unname(lapply(
     split(depends, depends$index), function(x) {
-      if (x$id_requested %in% c("latest", "latest()")) {
-        query <- sprintf('latest(name == "%s")', x$name)
-      } else if (grepl("latest\\(", x$id_requested)) {
-        str <- sub(")$", sprintf(' && name == "%s")', x$name), x$id_requested)
+      if (x$id_requested[[1]] %in% c("latest", "latest()")) {
```
Comment (M-Kusumgar) [inline R/migrate.R:256]: this is alright since these are guaranteed to be the same id, right?

**Review by M-Kusumgar — APPROVED** [review]
Nice one!

File: R/migrate.R (hunk 1 above)
Comment (richfitz) [reply R/migrate.R:252]: one as latest, the other as a fixed id?

Comment (richfitz) [reply R/migrate.R:252]: (that said, I think this is only present in very old reports)

**Review by richfitz — APPROVED** [review]

**Pull request merged**